### PR TITLE
Correcting Dataset Shift in Machine Learning

### PR DESCRIPTION
**Pull request by collins-a**
Article on correcting dataset shift in ML as index.md.
Hero image as hero.jpg.
Other images include covariate.png, covariate2.png, and prior2.png

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by collins-a** [comment]
This fixes #688

**Comment by hectorkambow** [comment]
plagiarism check done

File: articles/correcting-data-shift/index.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+2. Causes of dataset shift
+
+3. Types of dataset shift
+
+4. Correcting dataset shift
+
+### Prerequisites
+
+A general understanding of machine learning is required. This [post](Supervised Learning Algorithms | Section) can provide an introduction or refresher.
```
Comment (lalith1403) [inline articles/correcting-data-shift/index.md:18]: This link is invalid. Check this once.

File: articles/correcting-data-shift/index.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+Prior Probability Shift
+
+[Source](http://www.acad.bg/ebook/ml/The.MIT.Press.Dataset.Shift.in.Machine.Learning.Feb.2009.eBook-DDU.pdf)
+
+We can use the context of spam emails to better understand this type of shift. The contents of spam mail may vary over time. Their ratio to the total daily received email number may also vary over time. As a result, spam filters that were developed on emails received last month may not be quite as effective today as a result of a change in the composition of the traffic of the emails.
+
+Still, in the context of spam filters, let’s have an example of an unbalanced dataset. Consider a training set with equal prior probabilities on the count of received spam mail. If the prior probabilities are equal, the training set would contain 50% spam and 50% non-spam mail. However, if in actuality about 75% of our mail is spam, the prior probability of the class variables has been altered. The effect of this does not influence the input distribution, but rather the output distribution.
+
+For more on this type of shift, check out this [book]( Dataset Shift in Machine Learning (Neural Information Processing) (acad.bg)).
```
Comment (lalith1403) [inline articles/correcting-data-shift/index.md:68]: LINK IS INVALID

File: articles/correcting-data-shift/index.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+A minimax approach is used in adversarial search. In game theory, a minimax approach refers to a decision-making scheme that is used to maximize chances of victory and minimize the possibility of loss.
+
+In this context, adversarial search defines an approach where an adversarial model is used where the learning algorithm tries to build a predictor resilient to deletion of features during testing. The goal of adversarial search is to find the best minimax approach. This is by considering an adversary that deletes features, to find the best strategy. As is the nature of adversarial search, we note that one model attempts to maximize its score while the other one opposes it.
+
+#### Importance Reweighting
+
+When you have training examples that are very similar to test examples, it is necessary to up-weight the training examples. This makes it seem like the training data was taken from the test set. Each of the training instances is reweighted by the relative probability of the training set and test set. This can be done through density ratio estimation.
```
Comment (lalith1403) [inline articles/correcting-data-shift/index.md:98]: What do you mean by up weight the training examples? Can you explain more?

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Minor improvements and this should be ready for the final round of review. After editing, check for errors on Grammarly to ensure minimal errors.

**Comment by collins-a** [comment]
@lalith1403 as usual, thank you for your feedback. Cheers!

**Review by lalith1403 — APPROVED** [review]

**Comment by hectorkambow** [comment]
@collins-a  🚀  
Thank you for the great article - just finishing up my final review. 
Please be sure to go over the edits to see where we can continue to improve the overall readability of the aritcles. 👍

**Comment by collins-a** [comment]
@hectorkambow noted sir. Thank you

**Pull request merged**